block.go

```diff
@@ -469,8 +469,9 @@ Outer:
 }
 
 // CleanTombstones will rewrite the block if there any tombstones to remove them
-// and returns if there was a re-write.
-func (pb *Block) CleanTombstones(dest string, c Compactor) (bool, error) {
+// and returns a bool if there was a re-write. If there was a re-write,
+// it also returns the ulid of the new directory created.
```

Comment:
The doc comment doesn't make much sense to me. It says the method will rewrite the block, then it says it returns true _if_ there was a rewrite, implying in doesn't actually do this.

Also seems like this can simply be solved by returning the ULID of the result block, which may be the same one.

I know this isn't part of this PR, but I've issues with this API in general. Blocks are immutable, so any mutating function does not make sense to begin with. More so if it actually creates a new block, which makes the block object completely invalid after it returns. And why would I have to pass in a compactor for removing a subset of data from a single block?

@gouthamve

Reply:
AFAICS, it also doesn't re-read the meta file either, which makes the block object actually hold wrong data.